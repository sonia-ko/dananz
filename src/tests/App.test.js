import { render, screen, waitFor } from "@testing-library/react";
import App from "@/pages/_app";
import Home from "@/pages";
import { useRouter } from "next/router";

// Mock useRouter hook with custom implementation

// Define a mock AppProps object
const mockAppProps = {
  Component: jest.fn(),
  pageProps: {}, // You can customize this to match your specific test scenario
  router: jest.mock("next/router", () => ({
    useRouter: jest.fn(() => ({
      route: "/",
      pathname: "/",
      query: {},
      asPath: "/",
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(),
      prefetch: jest.fn(),
    })),
  })),
};

describe("App", () => {
  it("renders homepage unchanged", () => {
    const { container } = render(<App {...mockAppProps} />);
    console.log(container);
    expect(container).toMatchSnapshot();
  });
});

// describe("Page", () => {
//   it("renders an h1 element with the text 'Design your interior with high quality.'", () => {
//     // Define a mock router object
//     const mockRouter = {
//       pathname: "/",
//       push: jest.fn(),
//       prefetch: () => Promise.resolve(),
//       query: {},
//       asPath: "",
//     };

//     render(
//       <Router router={mockRouter}>
//         <Home {...mockAppProps} />
//       </Router>
//     );

//     // Check if an h1 element exists in the document
//     const heading = screen.getByRole("heading", {
//       name: /design your interior with high quality/i,
//     });

//     // Assert that the h1 element is present in the document
//     expect(heading).toBeInTheDocument();
//     // Assert that the h1 element has the correct text content
//     expect(heading).toHaveTextContent(
//       "Design your interior with high quality."
//     );
//   });
// });
