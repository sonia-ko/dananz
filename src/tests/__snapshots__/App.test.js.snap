// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders homepage unchanged 1`] = `<div />`;
